test/index.js

```diff
@@ -474,4 +475,156 @@ describe('HauteCouture', () => {
         });
     });
 
+    it('allows options to be optional', (done, onCleanup) => {
+
+        onCleanup((next) => {
+
+            reset();
+            return next();
+        });
+
+        const server = new Hapi.Server();
+        server.connection();
+
+        const noOptionsPlugin = (server, options, next) => {
+
+            HauteCouture()(server, (err) => {
+
+                server.app.realm = server.realm;
+                if (err) {
+                    return next(err);
+                }
+                next();
+            });
+        };
+
+        noOptionsPlugin.attributes = { name: 'no-options-plugin' };
+
+        server.register(noOptionsPlugin, (err) => {
+
+            if (err) {
+                return done(err);
+            }
+
+            // Just an example to show it used the caller's directory
+            expect(server.registrations['no-options-plugin']).to.exist();
+            done();
+        });
+    });
+
+    it('supports promises.', (done, onCleanup) => {
+
+        onCleanup((next) => {
+
+            reset();
+            return next();
+        });
+
+        const server = new Hapi.Server();
+        server.connection();
+
+        const plugin = (server, options, next) => {
+
+            HauteCouture()(server, options)
+            .then(() => {
+
+                server.app.realm = server.realm;
+                next();
+            })
+            .catch((err) => {
+
+                next(err);
+            });
+        };
+
+        plugin.attributes = { name: 'promise-plugin' };
+
+        server.register(plugin, (err) => {
+
+            if (err) {
+                return done(err);
+            }
+
+            // Just an example to show it used the caller's directory
+            expect(server.registrations['promise-plugin']).to.exist();
```

Comment:
This assertion doesn't actually test that haute-couture did anything aside from not fail when using promises.  A simple assertion, like proving that it registered `vision` would be useful.